docs/reference/analysis/tokenizers/classic-tokenizer.asciidoc

```diff
@@ -1,19 +1,94 @@
 [[analysis-classic-tokenizer]]
 === Classic Tokenizer
 
-A tokenizer of type `classic` providing grammar based tokenizer that is
-a good tokenizer for English language documents. This tokenizer has 
+The `classic` tokenizer provides a grammar based tokenizer that is
+a good tokenizer for English language documents. This tokenizer has
 heuristics for special treatment of acronyms, company names, email addresses,
-and internet host names. However, these rules don't always work, and 
-the tokenizer doesn't work well for most languages other than English.
-
-The following are settings that can be set for a `classic` tokenizer
-type:
-
-[cols="<,<",options="header",]
-|=======================================================================
-|Setting |Description
-|`max_token_length` |The maximum token length. If a token is seen that
-exceeds this length then it is discarded. Defaults to `255`.
-|=======================================================================
+and internet host names. However, these rules don't always work, and
+the tokenizer doesn't work well for most languages other than English:
+
+* It splits words at most punctuation characters, removing punctuation. However, a
+  dot that's not followed by whitespace is considered part of a token.
+
+* It splits words at hyphens, unless there's a number in the token, in which case
+  the whole token is interpreted as a product number and is not split.
+
+* It recognizes email addresses and internet hostnames as one token.
+
+[float]
+=== Example output
+
+[source,js]
+---------------------------
+POST _analyze
+{
+  "tokenizer": "classic",
+  "text": "The 2 QUICK Brown-Foxes jumped over the lazy dog's bone."
+}
+---------------------------
+// CONSOLE
+
+The above sentence would produce the following terms:
+
+[source,text]
+---------------------------
+[ The, 2, QUICK, Brown, Foxes, jumped, over, the, lazy, dog's, bone ]
+---------------------------
+
```

Comment:
I wonder if we could do something with `// TESTRESPONSE` here. I mean, we don't need to give the user the full output of the _analyze API but it'd be nice to actually assert the result somehow.